### PR TITLE
feat: handle Zero token price in UI

### PR DESCRIPTION
**Pull request by domw30**
[Associated Notion Card](https://www.notion.so/zerotech/dApp-Projects-e96437225e264ae8ae8d46ca5f4d7b35?p=d4265c99f36c4bad9f2c4bff392bd779)

-----

## 1. Pull request checklist

- [x] Notion card has been moved to the Code Review column
- [x] Notion card has a link to this PR
- [x] A reviewer has been assigned to the Notion card

## 2. PR type
- [x] feature

## 3. What is the old behaviour?
Wilder World token price widget was being displayed at root level, rather than the zero token price widget.

## 4. What is the new behaviour?
When at root level the zero token price widget will be displayed. When at Wilder World domain level the wilder world token price widget will be displayed.

Zero 
Screenshot 2022-05-09 at 14 37 30

Wild
Screenshot 2022-05-09 at 14 37 41

### CODE REVIEW

File: src/containers/other/PriceWidget/PriceWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+		zeroPercentageChange,
+	} = useCurrency();
+
+	const isMounted = useRef<boolean>();
+
+	//- Token Data
+	const [data, setData] = useState<TokenType>();
+
+	const getTokenData = useCallback(() => {
+		if (isRoot) {
```
Comment (domw30) [inline src/containers/other/PriceWidget/PriceWidget.tsx:49]: Once there are more subdomains like wilder world these conditions will need to be updated to be more generic.

**Comment by domw30** [comment]
@colbr, ready for review

File: src/containers/other/PriceWidget/PriceWidget.tsx [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+			{currentPriceDetails(Size.SML, data?.tokenTitle)}
+		</div>
+	);
+
+	const dropdownContent = (
+		<div
+			data-testid={selectors.dropDownContent}
+			className={`${styles.DropdownContent}`}
+		>
+			<div className={`${styles.Section} ${styles.SectionBaseline}`}>
```
Comment (colbr) [inline src/containers/other/PriceWidget/PriceWidget.tsx:183]: nit: you can use the `classnames` library for combined class names, e.g.:

```jsx
import classNames from 'classnames';
...
<div className={classNames(styles.Section, styles.SectionBaseLine)}></div>
...
```

You can also handle conditional classes like:

```jsx
import classNames from 'classnames/bind';
const cx = classNames.bind(styles);
...
<div className={cx(styles.Container, { Visible: isVisible })></div>
...
```

Not necessary for approval though!

**Review by colbr — APPROVED** [review]
Looks good to me. Will get @eric-zero-dev to review the redux/saga changes, and @EthFish to review the design!

File: src/store/currency/actionTypes.ts [hunk 3]
```diff
@@ -12,6 +12,13 @@ export const GET_LOOT_PRICE_USD_REQUEST = 'GET_LOOT_PRICE_USD_REQUEST';
 export const GET_LOOT_PRICE_USD_SUCCESS = 'GET_LOOT_PRICE_USD_SUCCESS';
 export const GET_LOOT_PRICE_USD_ERROR = 'GET_LOOT_PRICE_USD_ERROR';
 
+/**
+ * GET_ZERO_PRICE_USD action types
+ */
+export const GET_ZERO_PRICE_USD_REQUEST = 'GET_ZERO_PRICE_USD_REQUEST';
+export const GET_ZERO_PRICE_USD_SUCCESS = 'GET_ZERO_PRICE_USD_SUCCESS';
+export const GET_ZERO_PRICE_USD_ERROR = 'GET_ZERO_PRICE_USD_ERROR';
+
```
Comment (eric-zero-dev) [inline src/store/currency/actionTypes.ts:21]: Nice!

File: src/store/currency/actionTypes.ts [hunk 4]
```diff
@@ -21,3 +28,13 @@ export const GET_WILD_PRICE_PERCENTAGE_CHANGE_SUCCESS =
...
+/**
+ * GET_ZERO_PRICE_PERCENTAGE action types
+ */
+export const GET_ZERO_PRICE_PERCENTAGE_CHANGE_REQUEST =
+	'GET_ZERO_PRICE_PERCENTAGE_CHANGE_REQUEST';
+export const GET_ZERO_PRICE_PERCENTAGE_CHANGE_SUCCESS =
+	'GET_ZERO_PRICE_PERCENTAGE_CHANGE_SUCCESS';
+export const GET_ZERO_PRICE_PERCENTAGE_CHANGE_ERROR =
+	'GET_ZERO_PRICE_PERCENTAGE_CHANGE_ERROR';
```
Comment (eric-zero-dev) [inline src/store/currency/actionTypes.ts:40]: Nice 💪

File: src/store/currency/sagas.ts [hunk 5]
```diff
@@ -33,6 +46,29 @@ export function* getWildPriceUsdSaga() {
 	yield takeEvery(GET_WILD_PRICE_USD_REQUEST, getWildPriceUsd);
 }
 
+/**
+ * Get Wild Price 24hr Percentage Change USD saga
+ */
+export function* getWildPricePercentageChange() {
+	try {
+		const percentage: number = yield call(wildPricePercentageChange);
+		yield put(getWildPricePercentageChangeSuccess(percentage)); //
```
Comment (eric-zero-dev) [inline src/store/currency/sagas.ts:55]: nit: can we remove unused `//` at the end of this line?

File: src/store/currency/types.ts [hunk 6]
```diff
@@ -75,6 +109,12 @@ export type CurrencyActions =
...
+	| GetZeroPriceUsdRequest
+	| GetZeroPriceUsdSuccess
+	| GetZeroPriceUsdError
 	| GetWildPricePercentageChangeRequest
 	| GetWildPricePercentageChangeSuccess
-	| GetWildPricePercentageChangeError;
+	| GetWildPricePercentageChangeError
+	| GetZeroPricePercentageChangeRequest
+	| GetZeroPricePercentageChangeSuccess
+	| GetZeroPricePercentageChangeError;
```
Comment (eric-zero-dev) [inline src/store/currency/types.ts:120]: Nice 💪

**Review by eric-zero-dev — COMMENTED** [review]
Looks good and nice work @domw30. One small change is required with removing unused `//` in several places in redux part.

Also I recommend to add unit test for the `store/currency` 
- actions.test.ts
- reducer.test.ts
- saga.test.ts
- selector.test.ts (it is not needed the change since we don't have selector changes)

Thank you. Go Team 💪 💪 💪

**Review by eric-zero-dev — APPROVED** [review]
Great working @domw30 Thank you!

**Pull request merged**